lift/src/main/scala/com/linkedin/lift/mitigation/EOppUtils.scala

```diff
@@ -0,0 +1,156 @@
+    * @return the eCDF as a scala map
+    */
+  def cdfTransformation(data: DataFrame, probabilities: Array[Double],
+    relativeTolerance: Double): Map[Double, Double] = {
+    val quantiles = data.stat.approxQuantile("score", probabilities, relativeTolerance)
+    return quantiles.zip(probabilities).toMap
+  }
+
+  /**
+    * Adjust transformation such that the transformed score distribution is the same as the baseline score distribution
+    *
+    * @param spark spark session
+    * @param baselineData dataset containing score, label, attribute
+    * @param attributeList list of attributes
+    * @param transformations transformations represented as a scala Maps
+    * @param numQuantiles the number of quantiles for computing a quantile-quantile map between the original score
+    *                     and the transformed score
+    * @param relativeTolerance relative tolerance for computing approximate quantiles
+    * @return modified transformations
+    */
+  def adjustScale(spark: SparkSession, baselineData: Dataset[ScoreWithAttribute], attributeList: List[String],
+    transformations: Map[String, Map[Double, Double]],
+    numQuantiles: Int, relativeTolerance: Double): Map[String, Map[Double, Double]] = {
+
+    import spark.implicits._
+
+    val filteredData = baselineData
+      .filter($"attribute".isin(attributeList: _*))
+      .as[ScoreWithAttribute]
+
+    val probabilities = Array.range(0, numQuantiles + 1).map(x => x.toDouble / numQuantiles)
+
+    val quantilesBeforeTransformation = filteredData.stat.approxQuantile("score", probabilities,
+      relativeTolerance)
+    val transformedData = applyTransformation(spark, filteredData, attributeList, transformations)
+
+    val quantilesAfterTransformation = transformedData.stat.approxQuantile("score", probabilities,
+      relativeTolerance)
+
+    val qqMap = quantilesAfterTransformation.zip(quantilesBeforeTransformation).toMap
+
+    transformations.transform((attribute, innerMap) => innerMap.transform((key, value) =>
+      transformScore(value, quantilesAfterTransformation.toList, qqMap)))
+  }
+
+  /**
+    * Learning the equality of opportunity (EOpp) transformation for datasets
+    *
+    * @param spark             spark session
+    * @param data              dataset containing score, label, attribute
+    * @param numQuantiles      number of points for representing transformation functions
+    *                          (quantile-quantile mappings).
+    * @param relativeTolerance relative tolerance for computing approximate quantiles
+    * @param originalScale     whether the distribution of the transformed score should be the same as the distribution
+    *                          before transformation
+    * @return EOpp transformation represented as a scala Map[Double, Double] for each attribute
+    */
+  def eOppTransformation(spark: SparkSession, data: Dataset[ScoreWithLabelAndAttribute], attributeList: List[String],
```

Comment:
You can replace all usages of `List` with `Seq`.

Reply:
done. thanks